### PR TITLE
Locallyhostedmodels

### PR DESCRIPTION
**Pull request by spencer-crook**
added functionality for locally hosted models on macbook, needs to be cleaned up. please help.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -165,7 +212,7 @@ def surveycb():
                     label="Connection established!", state="complete", expanded=False
                 )
 
-    seed = st.text_input("Seed", value=42)
+    seed = st.number_input("Seed", value=0)
```
Comment (spencer-crook) [inline app.py:215]: Had to change str to int for ollama

File: app.py [hunk 2]
```diff
@@ -59,20 +63,50 @@ def create_answer(prompt):
             moderationfilter=moderationfilter,
             onlyusecontext=onlyusecontext,
         )
+        local = st.session_state.get("local", False)  # Default to False if not set
+        print(f"Local: {local}")
+        # Check if we are using a local model
+        if local:
```
Comment (spencer-crook) [inline app.py:69]: This logic should maybe be moved to models.py as I had to add an additional import here.

File: app.py [hunk 3]
```diff
@@ -42,6 +43,9 @@
 
 
 def create_answer(prompt):
+    # trying to capture model session state
+    st.session_state.orchestrator.load_secrets(model)
```
Comment (spencer-crook) [inline app.py:47]: I added this load_secrets() which was the behaviour of the test model button. Previously the model would not update until you hit the test connection button

**Review by bing1100 — APPROVED** [review]
Changes look good to push forward. We can make bug fixes in later patches

**Pull request merged**